internal/store/group.go

```diff
@@ -182,6 +169,89 @@ func (sqlStore *SQLStore) GetGroupRollingMetadata(groupID string) (*GroupRolling
 	return metadata, nil
 }
 
+// GetGroupStatus returns total number of installations in the group as well as number or
+// Installations already rolled out and awaiting rollout.
+//
+// Note: This function uses the same conditions as GetGroupRollingMetadata to be more accurate
+// with the internal state seen by the Group Supervisor.
+func (sqlStore *SQLStore) GetGroupStatus(groupID string) (*model.GroupStatus, error) {
+	group, err := sqlStore.GetGroup(groupID)
+	if err != nil {
+		return nil, err
+	}
+	if group == nil {
+		return nil, nil
+	}
+
+	var toBeRolledResult countResult
+	err = sqlStore.queryInstallationsToBeRolledOut(
+		[]string{"Count (*)"},
+		group,
+		&toBeRolledResult,
+	)
+	if err != nil {
+		return nil, err
+	}
+	installationsToBeRolled, err := toBeRolledResult.value()
+	if err != nil {
+		return nil, err
+	}
+
+	var rolledOutResult countResult
+	installationBuilder := sq.
+		Select("Count (*)").
+		From("Installation").
+		Where("GroupID = ?", group.ID).
+		Where("GroupSequence = ?", group.Sequence).
+		Where("State = ?", model.InstallationStateStable).
+		Where("DeleteAt = 0")
+	err = sqlStore.selectBuilder(sqlStore.db, &rolledOutResult, installationBuilder)
+	if err != nil {
+		return nil, err
+	}
+	rolledOutInstallations, err := rolledOutResult.value()
+	if err != nil {
+		return nil, err
+	}
+
+	totalInstallations, err := sqlStore.countInstallationsInGroup(group)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
We could save ourselves one query here. We can just take the total installation count and subtract installationsToBeRolled to get rolledOutResult for example. If we keep all three queries maybe we want to just double check the counts to make sure they add up and then log a warning if something went wrong?

Reply:
I might be wrong on that but I am not sure if that is correct. From what I understood from the existing code:
- rolled out installations are those:
  - with the latest group sequence
  - with stable state
- awaiting roll out are those:
  - not with the latest group sequence
  - with stable state

This leaves room for all other installations that are being currently updated/created/deleted. Those are included in the total count but are included neither in RolledOut nor AwaitingRollOut.
I think we can latter expand data returned by this endpoint to also include a count of installations in other states.